src/frontend/src/index.ts

```diff
@@ -104,17 +105,24 @@ const init = async () => {
   const addDeviceAnchor = getAddDeviceAnchor();
   if (addDeviceAnchor !== undefined) {
     // Register this device (tentatively)
-    await registerTentativeDevice(addDeviceAnchor, connection);
-
-    const i18n = new I18n();
-    const copy = i18n.i18n(copyJson);
-
-    // Show a good bye message
-    showMessage({
-      message: copy.close_page_device_added,
-      role: "notify-device-added",
-    });
-    return;
+    const { userNumber, alias: deviceAlias } = await registerTentativeDevice(
+      addDeviceAnchor,
+      connection
+    );
+
+    // Display a success page once device added (above registerTentativeDevice **never** returns if it fails)
+    await addDeviceSuccess({ deviceAlias });
+
+    // If user "Click" continue in success page, proceed with authentication
+    const result = await authenticate(connection, userNumber);
+    const { success } = await handleLoginFlowResult(result);
+
+    // User have successfully signed-in we can jump to manage page
+    if (nonNullish(success)) {
+      const { connection: authenticatedConnection } = success;
+      await manage({ userNumber, authenticatedConnection });
```

Comment:
it looks like `manage` is this:

``` ts
  // Here, if the user doesn't have any recovery device, we prompt them to add
  // one. The exact flow depends on the device they use.
  await recoveryWizard(userNumber, authenticatedConnection);
  // From here on, the user is authenticated to II.
  void renderManage(userNumber, authenticatedConnection);
```

would it make sense to simply call `renderManage`? if you've just added a device, you probably don't care about adding a recovery just now and you just want to complete the flow

Reply:
No idea, don't know the flow and routing enough well. Should I make this change in this PR or would you like to implement it yourself afterwards?

Reply:
I think you can do it in this PR. You can simply call `renderManage` instead of the new `manage` you created. Probably less surprising/annoying for the user!